_ingest-pipelines/processors/dissect.md

```diff
@@ -0,0 +1,289 @@
+`+` | Append | left | `%{keyname} %{+keyname}` | Appends two or more fields together. | 
+`+` with `/n` | Append with order | left and right | `%{+keyname}/2 %{+keyname/1}` | Appends two or more fields together in the order specified. |
+`?` | Named skip key | left | `%{?skipme}` | Skips the matched value in the output. Same behavior as `%{}`. |
+`*` and `&` | Reference keys | left | `%{*r1} %{&r1}` | Sets the output key as value of `*` and output value of `&`. |
+
+Detailed descriptions of each key modifier are in the following sections.
+
+### Right padding modifier (`->`)
+
+The dissection algorithm is precise and requires that every character in the pattern exactly match the source string. For instance, the pattern `%{helloworldkey} %{worldkey}` (one space) will match the string "Hello world" (one space) but not the string "Hello  world" (two spaces) because pattern only has one space while the source string has two.
+
+The right padding modifier can be used to address this issue. By adding the right padding modifier to the pattern `%{helloworldkey->} %{worldkey}`, it will no match `Hello world` (one space), `Hello  world` (two spaces), and even `Hello          world` (ten spaces). 
+
+The right padding modifier is used to allow for the repetition of characters following a `%{keyname->}`. The right padding modifier can be applied to any key along with any other modifiers. It should always be the rightmost modifier, for example, `%{+keyname/1->}`, `%{}`.
+
+#### Example of usage
+
+The following is an example of how to use a right padding modifier:
+
+`%{name->} %{city}, %{state} %{zip}`
+
+In this pattern, the right padding modifier `->` is applied to the `%{name}` key. This means that the `%{name}` key will match an sequence of characters, including spaces. This is useful for handling names that may contain spaces, such as "First Last".
+
+The following is an example of how the right padding would be used to extract information from the following address entries: 
+
+```bash
+New York, NY 10017
+New York City, NY 10017
+```
+
+Both addresses contain the same information, but the second entry has an extra word, `City`, in the city field. The right padding modifier allows the pattern to match both of these address entries, even though they have slightly different formats.
+
+### Append modifier (`+`)
+
+The append modifier combines the values of two or more keys into a single output value. The values are appended from left to right. You can also specify an optional separator to be inserted between the values. 
+
+#### Example of usage
+
+The following is an example of how to use the append modifier. In this example, the pattern extracts the values of `key1` and `key2` fields and appends them together, with a space as the separator:
```

Comment:
The field name to which you are appending has to be the same for all. Remove the `key1` and `key2` example and add the following example:

```
PUT /_ingest/pipeline/dissect-test
{
  "description": "Pipeline that dissects web server logs",
  "processors": [
    {
      "dissect": {
        "field": "message",
        "pattern": "%{+address}, %{+address} %{+address}",
        "append_separator": "|"
      }
    }
  ]
}
```
Test pipeline:
```
POST _ingest/pipeline/dissect-test/_simulate
{
  "docs": [
    {
      "_index": "testindex1",
      "_id": "1",
      "_source": {
        "message": "New York, NY 10017"
      }
    }
  ]
}
```
In the response, the substrings are appended into the `address` field:

```
{
  "docs": [
    {
      "doc": {
        "_index": "testindex1",
        "_id": "1",
        "_source": {
          "address": "New York|NY|10017",
          "message": "New York, NY 10017"
        },
        "_ingest": {
          "timestamp": "2024-01-22T22:30:54.516284637Z"
        }
      }
    }
  ]
}
```

Reply:
Thank you for the example.